presto-prometheus/src/test/java/io/prestosql/plugin/prometheus/TestPrometheusIntegrationTests3.java

```diff
@@ -73,7 +73,7 @@ public void testRetrieveUpValue()
         if (tries == maxTries) {
             assertTrue(false, "Prometheus container not available for metrics query in " + maxTries * timeBetweenTriesMillis + " milliseconds.");
         }
-        Thread.sleep(30 * 1000); // allow time to pass so Prometheus metric `up` has some data, should record at least one value, since every 15s is default
+        Thread.sleep(60 * 1000); // allow time to pass so Prometheus metric `up` has some data, should record at least one value, since every 15s is default
```

Comment:
I suggest polling the prometheus API directly on the `up` metric until you get results.  Configure a timeout where the test fails when it take's too long for the results to exist. Use the presto connector once the regular API returns results.

Reply:
I added an API calling check as well. In the end it was the TimeMachine code. Since other tests were setting now(), I needed to set that in these new tests.